packages/cli/src/constructs/api-assertion.ts

```diff
@@ -4,35 +4,96 @@ import {
   GeneralAssertionBuilder,
 } from './internal/assertion'
 
+/** Sources that can be used for API check assertions */
 type ApiAssertionSource =
   | 'STATUS_CODE'
   | 'JSON_BODY'
   | 'HEADERS'
   | 'TEXT_BODY'
   | 'RESPONSE_TIME'
 
+/**
+ * Assertion configuration for API checks.
+ * Allows validation of different aspects of HTTP responses.
+ */
 // Called Assertion instead of ApiAssertion for historical reasons.
 export type Assertion = CoreAssertion<ApiAssertionSource>
 
+/**
+ * Builder class for creating API check assertions.
+ * Provides convenient methods to create assertions for different parts of HTTP responses.
+ * 
+ * @example
+ * ```typescript
+ * // Status code assertions
+ * AssertionBuilder.statusCode().equals(200)
+ * AssertionBuilder.statusCode().greaterThan(199)
+ * AssertionBuilder.statusCode().lessThan(300)
+ * 
+ * // JSON body assertions using JSONPath
+ * AssertionBuilder.jsonBody('$.user.name').equals('John')
+ * AssertionBuilder.jsonBody('$.users.length').equals(5)
+ * AssertionBuilder.jsonBody('$.data[0].id').isNotNull()
+ * 
+ * // Header assertions
+ * AssertionBuilder.headers('content-type').contains('application/json')
+ * AssertionBuilder.headers('x-rate-limit-remaining').greaterThan(0)
+ * 
+ * // Text body assertions
+ * AssertionBuilder.textBody().contains('Welcome to our API')
+ * AssertionBuilder.textBody().notContains('error')
+ * 
+ * // Response time assertions
+ * AssertionBuilder.responseTime().lessThan(1000)
+ * AssertionBuilder.responseTime().greaterThan(100)
+ * ```
+ * 
+ * @see {@link https://jsonpath.com/ | JSONPath Online Evaluator}
```

Comment:
Let's scratch this. It might be unnecessary.

Reply:
I'll keep it since it seems like an important detail.